workflow-bot-app/src/main/java/com/symphony/bdk/workflow/management/WorkflowManagementService.java

```diff
@@ -101,14 +102,17 @@ public Optional<VersionedWorkflowView> get(String id, Long version) {
   }
 
   private VersionedWorkflow readAndValidate(Workflow workflow) {
-    Optional<VersionedWorkflow> activeVersion = versionRepository.findByWorkflowIdAndPublishedFalse(workflow.getId());
+    List<VersionedWorkflow> activeVersion = versionRepository.findByWorkflowId(workflow.getId());
     if (activeVersion.isEmpty()) {
       throw new NotFoundException(String.format(WORKFLOW_NOT_EXIST_EXCEPTION_MSG, workflow.getId()));
     }
-    if (activeVersion.get().getPublished()) {
-      throw new IllegalArgumentException("Update on a published Workflow is forbidden.");
+
+    if (activeVersion.stream().anyMatch(VersionedWorkflow::getPublished)) {
+      throw new UnsupportedOperationException(WORKFLOW_UPDATE_FORBIDDEN_EXCEPTION_MSG);
```

Comment:
I'm not sure what is the best Http code to return in this case. 405 does not look suitable as the Method itself is allowed but not on this resource. @ystxn

Reply:
sounds good